app/src/main/java/com/asfoundation/wallet/ui/iab/PaymentMethodsFragment.kt

```diff
@@ -133,7 +132,7 @@ class PaymentMethodsFragment : BasePageViewFragment(), PaymentMethodsView {
   override fun onAttach(context: Context) {
     super.onAttach(context)
     check(context is IabView) { "Payment Methods Fragment must be attached to IAB activity" }
-    iabView = context
+    iabView = context as IabActivity
```

Comment:
this cast Didn't break anything in the payment flow?

Reply:
Why need this cast?

Reply:
> this cast Didn't break anything in the payment flow?

You're right, it could brake, i'll leave it as it is, and add the new variable maybe

> Why need this cast?

I need to provide an activity to instantiate Offer Wall, and IabView is not an activity, but IabActivity, which extends IabView, is. So i changed iabView variable to be an IabActivity, so i can provide the activity to Offer Wall

Reply:
I forgot an extra step on the presenter, and now i don't need to do all of that, only provide the IabView to the presenter